command/debug/debug.go

```diff
@@ -320,212 +323,217 @@ func (c *cmd) captureStatic() error {
 	for output, v := range outputs {
 		marshaled, err := json.MarshalIndent(v, "", "\t")
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 
 		err = ioutil.WriteFile(fmt.Sprintf("%s/%s.json", c.output, output), marshaled, 0644)
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 	}
 
-	return errors
+	return errs
 }
 
-// captureDynamic blocks for the duration of the command
+// captureInterval blocks for the duration of the command
 // specified by the duration flag, capturing the dynamic
 // targets at the interval specified
-func (c *cmd) captureDynamic() error {
-	successChan := make(chan int64)
-	errCh := make(chan error)
+func (c *cmd) captureInterval() error {
+	intervalChn := time.NewTicker(c.interval)
+	defer intervalChn.Stop()
 	durationChn := time.After(c.duration)
 	intervalCount := 0
 
 	c.UI.Output(fmt.Sprintf("Beginning capture interval %s (%d)", time.Now().Local().String(), intervalCount))
 
-	// We'll wait for all of the targets configured to be
-	// captured before continuing
-	var wg sync.WaitGroup
+	err := captureShortLived(c)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case t := <-intervalChn.C:
+			intervalCount++
+			c.UI.Output(fmt.Sprintf("Capture successful %s (%d)", t.Local().String(), intervalCount))
+			err := captureShortLived(c)
+			if err != nil {
+				return err
+			}
+		case <-durationChn:
+			intervalChn.Stop()
+			return nil
+		case <-c.shutdownCh:
+			return errors.New("stopping collection due to shutdown signal")
```

Comment:
nit-pick: Do we consider this an error case? I'd assume if we are shutting down due to user request they are not expecting to see that reported as an error. But maybe this is consistent with prior behaviour. Not a big deal either way.

Reply:
the debug capture is user triggered, this will error when the application is stopped (by SIGINT or SIGTERM) while capturing debug logs. Could be useful in the case of an oom killer triggering the process to be killed.